src/app/platform/keys/create/page.tsx

```diff
@@ -1,21 +1,150 @@
+      );
+      setApiKey(response.apiKey);
+      setNotification("success");
+    } catch {
+      setNotification("error");
+    }
+  };
+
+  const handleCopy = () => {
+    navigator.clipboard.writeText(apiKey);
+    setCopied(true);
+    setTimeout(() => setCopied(false), 2000);
+  };
+
   return (
-    <div className="flex">
+    <div className="flex min-h-screen bg-gray-50">
       <Sidebar items={ApiKeyNavigations} />
-      <div className="flex-grow pl-4">
+      <div className="flex-grow p-8 max-w-4xl mx-auto">
         <MainHeader>
-          <KeyIcon className="h-12 w-12 inline-block" /> Create API Key
+          <div className="flex items-center">
+            <div className="p-3 bg-blue-100 rounded-lg">
+              <KeyIcon className="h-8 w-8 text-blue-600" />
+            </div>
+            <span className="ml-4 text-2xl font-bold text-gray-900">Create New API Key</span>
+          </div>
         </MainHeader>
+
+        <form onSubmit={handleSubmit} className="mt-8 space-y-8 bg-white rounded-xl shadow-sm border border-gray-200 p-8">
+          <div className="space-y-6">
+            <div>
+              <label className="block text-sm font-medium text-gray-700 mb-2">Name</label>
+              <input
+                type="text"
+                value={name}
+                onChange={(e) => setName(e.target.value)}
+                className="w-full px-4 py-3 rounded-lg border border-gray-300 focus:ring-2 focus:ring-blue-500 focus:border-transparent transition duration-200"
+                placeholder="e.g. API Key"
+                required
+              />
+            </div>
+
+            <div>
+              <label className="block text-sm font-medium text-gray-700 mb-2">Description</label>
+              <textarea
+                value={description}
+                onChange={(e) => setDescription(e.target.value)}
+                rows={3}
+                className="w-full px-4 py-3 rounded-lg border border-gray-300 focus:ring-2 focus:ring-blue-500 focus:border-transparent transition duration-200"
+                placeholder="What will this API key be used for?"
+                required
+              />
+            </div>
+
+            <div>
+              <label className="block text-sm font-medium text-gray-700 mb-2">Expiration Period</label>
+              <select
+                value={expiry}
+                onChange={(e) => setExpiry(e.target.value)}
+                className="w-full px-4 py-3 rounded-lg border border-gray-300 focus:ring-2 focus:ring-blue-500 focus:border-transparent transition duration-200"
+                required
+              >
+                <option value="">Select duration</option>
+                <option value="30">30 days</option>
+                <option value="60">60 days</option>
+                <option value="90">90 days</option>
+                <option value="365">1 year</option>
+                <option value="never">Never expires</option>
```

Comment:
Lets remove `Never expires` because our backend does not support anything more than 365